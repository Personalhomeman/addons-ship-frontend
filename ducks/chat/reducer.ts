import { Message, ChatActionTypes } from './types';
import { ChatActions } from './actions';

export interface ChatState {
  messages: Message[];
}

const initialState: ChatState = {
  messages: []
};

export const reducer = (state = initialState, action: ChatActions): ChatState => {
  switch (action.type) {
    case ChatActionTypes.SendMessage:
    case ChatActionTypes.ReceiveMessage:
      return {
        ...state,
        messages: [...state.messages, action.payload]
      };
    default:
      return state;
  }
};
