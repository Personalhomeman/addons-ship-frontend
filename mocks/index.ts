import { AppVersion } from '@/models';
import { TestDevice } from '@/models/test-device';

export const mockAppVersion: AppVersion = {
  id: 123,
  version: '1.0.3',
  platform: 'ios',
  buildNumber: 32,
  buildSlug: '13245ads',
  lastUpdate: new Date('2018-09-22T12:42:31Z'),
  description: 'Some semi-long description',
  whatsNew: 'Some longer whats-new',
  minimumOs: '10.2',
  minimumSdk: 'dunno',
  packageName: 'com.package.name',
  certificateExpires: new Date('2018-09-22T12:42:31Z'),
  bundleId: 'bundle-id',
  size: 4567034,
  supportedDeviceTypes: ['iphone', 'ipad'],
  distributionType: 'development',
  iconUrl: 'http://placekitten.com/160/160',
  appName: 'Standup Timer',
  promotionalText: 'Promotional Text',
  keywords: 'Keywords',
  reviewNotes: 'Review Notes',
  supportUrl: 'https://bitrise.io/support',
  marketingUrl: 'https://bitrise.io/marketing',
  scheme: 'piramid',
  configuration: 'canary',
  publicInstallPageURL: 'https://bitrise.io/app/8b334705d8e78276'
};

export const mockTestDevices: TestDevice[] = [
  { deviceId: 'some-device-id-01', deviceType: 'ios', owner: 'test-user-1' },
  { deviceId: 'some-device-id-02', deviceType: 'android', owner: 'test-user-1' },
  { deviceId: 'some-device-id-03', deviceType: 'ios', owner: 'test-user-2' }
];
