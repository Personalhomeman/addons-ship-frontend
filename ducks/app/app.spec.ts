jest.mock('@/services/ship-api');

import configureMockStore, { MockStoreCreator, MockStoreEnhanced } from 'redux-mock-store';
import thunk from 'redux-thunk';

import shipApi from '@/services/ship-api';
import { mockApp } from '@/mocks';

import reducer, { fetchApp, selectPlatform } from '.';

describe('app', () => {
  let mockStore: MockStoreCreator;
  beforeEach(() => {
    mockStore = configureMockStore([thunk.withExtraArgument({ shipApi })]);
  });

  describe('reducer', () => {
    it('loads an app', () => {
      const state = reducer(undefined, fetchApp.complete(mockApp));

      expect(state).toMatchSnapshot();
    });

    it('selects a platform', () => {
      const state = reducer(undefined, selectPlatform('ios'));

      expect(state).toMatchSnapshot();
    });
  });

  describe('fetchApp', () => {
    let store: MockStoreEnhanced;

    beforeEach(() => {
      store = mockStore();
    });

    it('fetches app version events', async () => {
      const appSlug = 'balatonon';
      (shipApi.getApp as jest.Mock).mockResolvedValueOnce(mockApp);
      await store.dispatch(fetchApp(appSlug) as any);

      expect(shipApi.getApp).toHaveBeenCalledWith(appSlug);
      expect(store.getActions()).toMatchSnapshot();
      expect(store.getState()).toMatchSnapshot();
    });

    it("can't fetch an app", async () => {
      (shipApi.getApp as jest.Mock).mockRejectedValueOnce('api had some issue');
      await store.dispatch(fetchApp('froccsozok') as any);

      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
