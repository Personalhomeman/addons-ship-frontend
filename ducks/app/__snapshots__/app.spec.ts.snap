// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app fetchApp can't fetch an app 1`] = `
Array [
  Object {
    "type": "APP_GET_NEXT",
  },
  Object {
    "payload": "api had some issue",
    "type": "APP_GET_ERROR",
  },
]
`;

exports[`app fetchApp fetches app version events 1`] = `
Array [
  Object {
    "type": "APP_GET_NEXT",
  },
  Object {
    "payload": Object {
      "appSlug": "test-app-slug-1",
      "avatarUrl": "https://concrete-userfiles-production.s3.us-west-2.amazonaws.com/repositories/0dbc45647ce84cb9/avatar/avatar.png",
      "plan": "gold",
      "projectType": "other",
      "title": "Awesome app",
    },
    "type": "APP_GET_COMPLETE",
  },
]
`;

exports[`app fetchApp fetches app version events 2`] = `Object {}`;

exports[`app reducer loads an app 1`] = `
Object {
  "app": Object {
    "androidErrors": Array [],
    "appSlug": "test-app-slug-1",
    "avatarUrl": "icon.url",
    "iosErrors": Array [],
    "plan": "gold",
    "projectType": "other",
    "title": "Awesome app",
  },
  "selectedPlatform": "ios",
}
`;

exports[`app reducer selects a platform 1`] = `
Object {
  "app": null,
  "selectedPlatform": "ios",
}
`;
